tests/unit/test_task_creation_signatures.py

```diff
@@ -0,0 +1,15 @@
+import inspect
+
+from procrastinate import blueprints
+
+
+def test_task_signatures(app, mocker):
+    """Tasks can be created in two ways, both of which need to maintain an
+    identical API.  This test simple test that App.task and Blueprint.task have
+    the same function signature.
+
+    This is further enforced with protocols.TaskCreator and mypy checks.
+    """
+
+    bp = blueprints.Blueprint()
+    assert inspect.signature(bp.task) == inspect.signature(app.task)
```

Comment:
Wait... If we make a protocol, why don't we use it for the test ?

Reply:
So I think you're suggesting we use `runtime_checkable` and a `isinstance`.
This would confirm that both `Blueprint` and `App` have a `task` method, but would not confirm anything about the `task` method signature.

As far as I understand, only mypy and `inspect.signature` can do this in depth check.

Reply:
```python
In [5]: @runtime_checkable
   ...: class Drivable(Protocol):
   ...:     def drive(self, from_place: str, to_place: str)->int:
   ...:         ...
   ...:

In [6]: class Car:
   ...:     def drive(self):
   ...:         return 0
   ...:

In [7]: isinstance(Car, Drivable)
Out[7]: True
```

This fills me with sadness but otherwise, you're absolutely right.

Reply:
Shouldn't we check that App signature  == protocol signature, and blueprint signature == protocol signature ?
Also I believe we could test that on the classes directly and not on the instances.

Reply:
> ... test that on the classes directly and not on the instances.

Good call.   Have updated.

Reply:
> Shouldn't we check that App signature == protocol signature, and blueprint signature == protocol signature

So interesting it appears that MyPy will ignore default values when comparing Protocol method signatrues, but our explicit test using `inspect` requires that defaults be identical.

I've update the Protocol.task signature to include defaults for the moment as this seems to be what we want.  But this might trip us up in the future.